### DISCUSSION TITLE
Integrating with my bot's site

### DISCUSSION THREAD

**Discussion by qharscoet** [Glitchii/embedbuilder#12] [Ideas]
Hi Glitchiii,

I was looking for an embed builder/visualizer simple enough and easy to use to integrate into my server's bot website to allow my non-programmers staff to send nice embeds.
What would be the best way to fetch the json from the builder ?

It seems like it has minimal dependencies and external files to get so that would be quite easy to just copy and paste it into my node app, but I kinda like the iframe integration as it's really seamless, keeps up with potentiel future updates and allows me to not add any additional files to my project (and the URL options available cover my needs just fine 👌). 
Problem is I can't really access the json variable inside the iframe from my website due to CORS. Do you have any way of getting around that ? I looked it up a little bit and it seems it can be done safely with the window.postMessage API or something. Do you have any plans to integrate this kind of stuff ? I can also look into it and probably can do a PR later if needed.

**Comment by zignis** [comment]
You can fork this repo and add a button, which when clicked posts a message with JSON content using `window.postMessage`, and catch it on your website's end. I would always try to avoid using iFrame as you can easily integrate the embed builder into your site and access the JSON using `editor.getValue()`.
